tests/lib/prepare.sh

```diff
@@ -44,20 +44,25 @@ update_core_snap_for_classic_reexec() {
     cp -a "$LIBEXECDIR"/snapd/* squashfs-root/usr/lib/snapd/
     # also the binaries themselves
     cp -a /usr/bin/{snap,snapctl} squashfs-root/usr/bin/
-    # and snap-confine's apparmor
-    if [ -e /etc/apparmor.d/usr.lib.snapd.snap-confine.real ]; then
-        cp -a /etc/apparmor.d/usr.lib.snapd.snap-confine.real squashfs-root/etc/apparmor.d/usr.lib.snapd.snap-confine.real
-    else
-        cp -a /etc/apparmor.d/usr.lib.snapd.snap-confine      squashfs-root/etc/apparmor.d/usr.lib.snapd.snap-confine.real
-    fi
+
+    case "$SPREAD_SYSTEM" in
+        ubuntu-*|debian-*)
+            # and snap-confine's apparmor
+            if [ -e /etc/apparmor.d/usr.lib.snapd.snap-confine.real ]; then
+                cp -a /etc/apparmor.d/usr.lib.snapd.snap-confine.real squashfs-root/etc/apparmor.d/usr.lib.snapd.snap-confine.real
+            else
+                cp -a /etc/apparmor.d/usr.lib.snapd.snap-confine      squashfs-root/etc/apparmor.d/usr.lib.snapd.snap-confine.real
+            fi
+            ;;
+    esac
 
     # repack, cheating to speed things up (4sec vs 1.5min)
     mv "$snap" "${snap}.orig"
     mksnap_fast "squashfs-root" "$snap"
     rm -rf squashfs-root
 
     # Now mount the new core snap, first discarding the old mount namespace
-    /usr/lib/snapd/snap-discard-ns core
+    $LIBEXECDIR/snapd/snap-discard-ns core
```

Comment:
For the purpose of writing tests I was thinking about `snap tool discard-ns core` that could aggregate all distro differences this way.